assets/src/scripts/components/hydrograph/timeseries.js

```diff
@@ -82,7 +82,22 @@ export const methodsSelector = state => state.series.methods;
 /**
  * @return {Object}     Mapping of time series ID to time series details
  */
-export const allTimeSeriesSelector = state => (state.series ? state.series.timeSeries : {}) || {};
+export const allTimeSeriesSelector = createSelector(
+    state => state.series,
+    (stateSeries) => {
+        let timeSeries = {};
+        if (stateSeries && stateSeries.hasOwnProperty('timeSeries')) {
+            let stateTimeSeries = stateSeries.timeSeries;
+            for (let key of Object.keys(stateTimeSeries)) {
+                const ts = stateTimeSeries[key];
+                if (ts.hasOwnProperty('points') && ts.points.length > 0) {
+                    timeSeries[key] = ts;
+                }
+            }
+        }
+        return timeSeries;
+    }
+);
```

Comment:
I'm running this locally and found an issue on this site: http://localhost:5050/monitoring-location/251549080251200

The problem seems to be that timeSeriesSelector assumes that all the series in the "collection" are returned from allTimeSeriesSelector.

Yesterday we were talking about how the collection and request selectors are confused. Maybe you could just get rid of them as a part of this? The `state.series.timeSeries` elements have a `tsKey` attribute that can be used to filter by current, compare, etc.

Reply:
@danielnaab: I might merge these changes in and remove those to selectors in a separate PR.

Reply:
👍